### PR TITLE
REST First draft

### LINKED ISSUE #122: REST

### LINKED ISSUE #199: REST Lab and Prep

### PR DESCRIPTION
**Pull request by rtillies**

### CODE REVIEW

File: module3/lessons/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/lessons/Week2/RestfulRoutes.md:11]: I think it could be helpful to do a review of CRUD here  - that way you can associate the routes with different crud actions.  Since we won't yet have a fully formed MVC app, students can still see how the different HTTP Methods match up to our standard Database actions.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/lessons/Week2/RestfulRoutes.md:27]: One thing we've done on BE to help re-inforce these routes is to have an activity where we give students a resource (like muffin) and have them list all the routes that would exist.  We could even gamify it - quiz show for resource/crud action and what should the route be.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md]: I'm not sure what these will look like on the live site, might totally work and in that case go for it! If it looks weird on the live site, this is the slightly different syntax I have been using: https://github.com/turingschool/launch-curriculum#answersolution-box

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md]: I really like having all the routes displayed in a table! Makes it easy to see all of the options.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md]: The language "Use form to edit existing movie" feels a bit confusing to me. Maybe "Display form to edit existing movie" might be clearer? Or some other language? I can see students being confused about the different between Edit and Update and trying to think how to make it clear /movies/:id/edit gives you the page with the form.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md:138]: I really like the flow of your lesson where you add the annotation at the end after students realize there is a need for it!

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md:119]: We might want to say a little bit of something about why we have @model Movie at the top and how the @ symbol is razor syntax for letting us write C# code in a razor file.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md]: This might be too nit, but I think we do want to try and be as precise as we can with terms. I think technically the second column is the `path` not the full `uri`. I like the images in the URI link that you sent a lot.

Screenshot 2023-05-18 104515

Comment (zoefarrell) [reply module3/lessons/Week2/RestfulRoutes.md]: In the post it says that these are all URIs: https://en.wikipedia.org/wiki/Uniform_Resource_Identifier

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md:173]: Random idea you can feel free to take or leave, might be interesting to show some examples of non-restful routes and restful routes and have students pick which are which.

**Comment by zoefarrell** [comment]
This looks great! I left some small comments, but I think will be a great introduction to RESTful routes. I would approve but I assume you'll add the prep and lab here so probably best to wait until someone reviews all the parts.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (rtillies) [reply module3/lessons/Week2/RestfulRoutes.md]: I agree!

File: module3/lessons/Week2/RestfulRoutes.md
Comment (rtillies) [reply module3/lessons/Week2/RestfulRoutes.md]: Fixed!

File: module3/lessons/Week2/RestfulRoutes.md
Comment (rtillies) [reply module3/lessons/Week2/RestfulRoutes.md:138]: Thanks!

File: module3/lessons/Week2/RestfulRoutes.md
Comment (rtillies) [reply module3/lessons/Week2/RestfulRoutes.md:119]: Done! I'd like your feedback for clarity and accuracy.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (rtillies) [reply module3/lessons/Week2/RestfulRoutes.md:11]: Learning objectives added, including a basic review of CRUD. Thanks for the feedback!

**Comment by zoefarrell** [comment]
Took another look, from my perspective, the lesson looks ready to merge! Let me know when you want me to take a look at the Lab/Prep too.

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md:129]: I like this, feels like a good amount of detail!

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [inline module3/lessons/Week2/RestfulRoutes.md]: Just realized that similar to my last question, this doesn't match the URI definition on Wikipedia where the entire thing is the URI. Might be helpful to include the image from Wikipedia here, it helps me visualize what is the URI.

Comment (zoefarrell) [reply module3/lessons/Week2/RestfulRoutes.md]: https://en.wikipedia.org/wiki/Uniform_Resource_Identifier

File: module3/labs/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/labs/Week2/RestfulRoutes.md:14]: Should we have a minimum number of records they should be creating? Like 3 albums minimum?

File: module3/labs/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/labs/Week2/RestfulRoutes.md:20]: Love that you've shown these error messages!!

File: module3/labs/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/labs/Week2/RestfulRoutes.md:45]: Yay, migrations practice!!

File: module3/preparation/Week2/RestfulRoutes.md
Comment (memcmahon) [inline module3/preparation/Week2/RestfulRoutes.md:12]: ✅

**Review by memcmahon — APPROVED** [review]
The prep and lab look great - I just left a note for one small suggestion. If you agree, I would say this is ready to merge!

File: module3/labs/Week2/RestfulRoutes.md
Comment (zoefarrell) [reply module3/labs/Week2/RestfulRoutes.md:45]: Wrote this on the day we were learning migrations in Mod 2 and realized we will definitely want some spiraling here 😆

File: module3/lessons/Week2/RestfulRoutes.md
Comment (zoefarrell) [reply module3/lessons/Week2/RestfulRoutes.md:27]: I like inserting an activity here! At least we could do some waterfalls for the different routes

**Pull request merged**
